src/pages/PromptPage.vue

```diff
@@ -66,128 +65,69 @@
 </template>
 
 <script setup>
-import { useQuasar } from 'quasar'
 import BarGraph from 'src/components/BarGraph.vue'
 import TheEntries from 'src/components/TheEntries.vue'
 import ShareComponent from 'src/components/ShareComponent.vue'
-import { usePromptStore, useStatStore, useUserStore } from 'src/stores'
+import { usePromptStore, useStatStore } from 'src/stores'
 import { onMounted, ref } from 'vue'
 import { useRouter } from 'vue-router'
 
-const $q = useQuasar()
 const router = useRouter()
 
 const promptStore = usePromptStore()
 const statStore = useStatStore()
-const userStore = useUserStore()
 
 const chartData = ref([])
 const prompt = ref({})
 const tab = ref('prompt')
-
 onMounted(async () => {
   statStore.fetchStats()
-
   if (router.currentRoute.value.href === '/month') {
     await promptStore.fetchMonthPrompt()
     prompt.value = promptStore.getMonthPrompt
+  } else {
+    await promptStore.fetchPrompts()
```

Comment:
Is it really necessary to call all the prompts here?
Were you getting any errors previously?

Reply:
If we don't call it here, when we reload the page we don't have any data

Reply:
Are you sure?

We have `await updatePrompt()` that checks the `href` and fetch it.

Reply:
I remove this function
in develop we use it like this 


```js
onMounted(async () => {
  statStore.fetchStats()
  if (router.currentRoute.value.href === '/month') {
    await promptStore.fetchMonthPrompt()
    prompt.value = promptStore.getMonthPrompt
  }
  if (promptStore.getPrompts.length) {
    promptStore.getPrompts.find((p) => {
      if (
        p.id === `${router.currentRoute.value.params.year}-${router.currentRoute.value.params.month}` ||
        p.slug === router.currentRoute.value.params.slug
      ) {
        prompt.value = p
        updateChartData()
      }
    })
    return
  }
// Here
  await updatePrompt()
  if (prompt.value) {
    updateChartData()
  }
})
....
async function updatePrompt() {
  if (router.currentRoute.value.params.year) {
    await promptStore
      .fetchPromptById(`${router.currentRoute.value.params.year}-${router.currentRoute.value.params.month}`)
      .then((res) => (prompt.value = res))
      .catch(() => router.push('/404'))
  }
  if (router.currentRoute.value.params.slug) {
    await promptStore
      .fetchPromptBySlug(router.currentRoute.value.params.slug)
      .then((res) => (prompt.value = res))
      .catch(() => router.push('/404'))
  }
  updateChartData()
}
function like() {

// Here
  promptStore.addLike(prompt.value.id).then(() => updatePrompt())
}
function dislike() {

// Here
  promptStore.addDislike(prompt.value.id).then(() => updatePrompt())
}
```

As you can see we check many time the routing to decide how the update should work. 

Now i make it simplier so we load the data one times, and we do not need it in the like and dislike function.